PWGLF/Tasks/spectraCharged.cxx

```diff
@@ -221,11 +221,11 @@ bool chargedSpectra::initParticle(const P& particle)
 {
   vars.isChargedPrimary = false;
   auto pdgParticle = pdg->GetParticle(particle.pdgCode());
-  // if (!pdgParticle || pdgParticle->Charge() == 0.) {
-  if (!pdgParticle || TMath::Abs(pdgParticle->Charge()) != 3.) { // This is only a temporary workaround for isPhysicalPrimary and should be replaced by commented line in future
+  if (!pdgParticle || pdgParticle->Charge() == 0.) {
```

Comment:
Here you are skipping neutral and those for which the PDG is not taken and this is fine

Reply:
> Hi @pestratm wouldn't it be better to iteratively check the mother? Something like https://github.com/AliceO2Group/DelphesO2/blob/master/examples/aod/createO2tables.h#L643 I'm not sure that you are selecting primaries like this, unless I misunderstood what you're doing

You're correct, it is not actually selecting primaries. It is more of a quick and "dirty" temporary workaround to only select particles that at least could be primary particles (and thus exclude other particles that are too short-lived to be primaries). I will change it again once isPrimaryParticle is properly working. This is why I only checked if it is one of those particles and not if it actually really is a primary